### PR TITLE
Server Side Apply: Enabled without detection logic when k8s version > v1.22

### LINKED ISSUE #37946: server-side apply detection logic does not work for upgraded clusters
### Bug Description

The server-side apply detection logic for istioctl only works for newly created 1.18 clusters. It doesn't work for clusters that were upgraded.

https://github.com/istio/istio/blob/dac4c97a792a6fc362165b6385429413d90793f1/operator/pkg/helmreconciler/reconciler.go#L251-L258

Our server does support server-side apply. But because the kube-system namespace was created under an older cluster version, it does not have the `managedFields` field.

This is partially the cause of #37912, due to an inconsistency between client- and server-side apply I filed here: https://github.com/kubernetes/kubernetes/issues/108708

### Version

```prose
$ istioctl version
client version: 1.11.6
control plane version: 1.11.6
data plane version: 1.11.6 (129 proxies)

$ kubectl version --short
Client Version: v1.20.15
Server Version: v1.20.11-eks-f17b81
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Pawan-Bishnoi**
CheckSSAEnabled for version > 1.22 should always return true since it went GA in v1.22

Fixes: #37946

### CODE REVIEW

**Review by ymesika — APPROVED** [review]
LG

**Comment by Pawan-Bishnoi** [comment]
@ymesika can you mark this /ok-to-test

**Comment by zirain** [comment]
> @ymesika can you mark this /ok-to-test

you should apply to join [community](https://github.com/istio/community) to skip this.
like this PR https://github.com/istio/community/pull/983

File: operator/pkg/helmreconciler/reconciler.go [hunk 1]
```diff
@@ -247,9 +247,9 @@ func (h *HelmReconciler) processRecursive(manifests name.ManifestMap) *v1alpha1.
 // CheckSSAEnabled is a helper function to check whether ServerSideApply should be used when applying manifests.
 func (h *HelmReconciler) CheckSSAEnabled() bool {
 	if h.kubeClient != nil {
-		// There is a mutatingwebhook in gke that would corrupt the managedFields, which is fixed in k8s 1.18.
-		// See: https://github.com/kubernetes/kubernetes/issues/96351
-		if kube.IsAtLeastVersion(h.kubeClient, 18) {
+		// SSA went GA in k8s 1.22. Earlier versions (eg: 1.18) have an issue where managedFields is not present on upgraded clusters.
+		// See: https://github.com/istio/istio/issues/37946#issuecomment-1072875625
+		if kube.IsAtLeastVersion(h.kubeClient, 22) {
```
Comment (howardjohn) [inline operator/pkg/helmreconciler/reconciler.go:252]: can we do `if >22 { return true } else if > 18 { detect } else { return false }`

Right now users on 1.18-1.21 may be broken by this change

File: releasenotes/notes/37946.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+releaseNotes:
+  - |
+    **Fixed** server-side apply detection logic to not run for k8s cluster version < 1.22
```
Comment (dhawton) [inline releasenotes/notes/37946.yaml:6]:
```suggestion
    **Fixed** server-side apply detection logic to not run for Kubernetes cluster versions below 1.22
```

**Review by howardjohn — APPROVED** [review]

File: releasenotes/notes/37946.yaml [hunk 3]
```diff
@@ -3,4 +3,4 @@ kind: bug-fix
 area: istioctl
 releaseNotes:
   - |
-    **Fixed** server-side apply detection logic to not run for k8s cluster version < 1.22
+    **Fixed** Server Side Apply is enabled by default for Kubernetes cluster versions above 1.22
```
Comment (dhawton) [inline releasenotes/notes/37946.yaml:6]:
```suggestion
    **Fixed** Server Side Apply to not run for Kubernetes cluster versions below 1.22
```

**Review by dhawton — CHANGES_REQUESTED** [review]

File: releasenotes/notes/37946.yaml (hunk 3 above)
Comment (Pawan-Bishnoi) [reply releasenotes/notes/37946.yaml:6]: Hi @dhawton 
After John's initial review, the logic was changed to:

```
if v>22
  ssa is enabled
else if v>18
  ssa detection logic is run
```

Comment (Pawan-Bishnoi) [reply releasenotes/notes/37946.yaml:6]: so SSA will be used for versions below 22 if it is detected (same behaviour as earlier)

**Comment by Pawan-Bishnoi** [comment]
Update on this:
It seems enabling ServerSideApply by default is breaking the tests. 
I see a lot of [this error](https://github.com/istio/istio/blob/f21f0dfe71914bc9f647713a2eaefdadc86b5228/operator/pkg/helmreconciler/apply.go#L228-L230) in the test logs. Checking how to make SSA work with our test setup.

File: operator/pkg/helmreconciler/reconciler.go (hunk 1 above)
Comment (Pawan-Bishnoi) [reply operator/pkg/helmreconciler/reconciler.go:254]: Do you think it is worth adding a reconciler option to disable SSA ? 🤔 
Can't see why someone would want to do it but checking anyways.

**Comment by Pawan-Bishnoi** [comment]
Disabled SSA while in TestMode. Please let me know if that is not the right way of moving forward.
https://github.com/istio/istio/blob/f21f0dfe71914bc9f647713a2eaefdadc86b5228/operator/pkg/helmreconciler/common.go#L53-L54

File: releasenotes/notes/37946.yaml (hunk 3 above)
Comment (dhawton) [reply releasenotes/notes/37946.yaml:6]: Fixed my suggestion based on that logic.

**Review by howardjohn — APPROVED** [review]

**Comment by Pawan-Bishnoi** [comment]
@dhawton it is waiting for your approval. Can you please review and let me know if there are additional comments.

**Review by dhawton — APPROVED** [review]

**Pull request merged**